client/src/controllers/MessagesController.ts

```diff
@@ -0,0 +1,46 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class MessagesController extends Controller<HTMLFormElement> {
+  static classes = ['newMessage', 'show'];
+  static targets = ['messages', 'templates'];
+
+  declare readonly hasMessagesTarget: boolean;
+  declare readonly hasNewMessageClass: boolean;
+  declare readonly hasShowClass: boolean;
+  declare readonly hasTemplatesTarget: boolean;
+  declare readonly messagesTarget: HTMLElement;
+  declare readonly newMessageClass: string;
+  declare readonly showClass: string;
+  declare readonly templatesTarget: HTMLElement;
+
+  add(event) {
+    const status = event.detail.status;
+    const statusText = event.detail.text;
+    // console.log("connected add.",status,statusText);
+    // console.log(this.templatesTarget);
+    if (this.hasNewMessageClass) {
+      this.element.classList.add(this.newMessageClass);
+    }
+    if (this.hasTemplatesTarget) {
+      const template = this.templatesTarget as HTMLTemplateElement;
+      const cloneTemplate = template.content.cloneNode(true) as HTMLElement;
```

Comment:
You should be able to simplify this a bit - give something like this a go.

```js
    // note: if you declare templatesTarget above with `declare templatesTarget: HTMLTemplateElement;` you should not need to use `as HTMLTemplateElement` here.
    const message = this.templatesTarget.content
      .querySelector(`[data-message-status="${status || 'success'}"]`)
      ?.cloneNode(true);

    if (!message) return;

  /// (message as HTMLElement).querySelector('span')... etc
```

* With a bit of digging, you can see that the `.content` returns a DocumentFragment - https://developer.mozilla.org/en-US/docs/Web/API/HTMLTemplateElement/content
* DoumentFragment allows querySelector to be used - https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment & https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment/querySelector
* From here we can clone that found Node to give us a potential `HTMLElement` to work with


Best to call this element you are generating something contextual to the code, like `message` or `messageElement`. The controller does not know it's going to generate a `list element`, that's defined in the HTML. This could work 100% the same if it's adding rows to a table or articles to a section.

Reply:
awesome! its working thanks a lot.